spring-boot-samples/spring-boot-sample-cache-ehcache/src/main/java/sample/Application.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2012-2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sample;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.cache.annotation.EnableCaching;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.stereotype.Component;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * @author Eddú Meléndez
+ * @since 1.3.0
+ */
+@EnableCaching
+@SpringBootApplication
+public class Application {
+
+	public static void main(String[] args) {
+		ConfigurableApplicationContext context = SpringApplication.run(Application.class);
+		LanguageService cachingService = context.getBean(LanguageService.class);
+		System.out.println(cachingService.languages());
+		System.out.println(cachingService.languages());
+	}
+
+}
+
+interface LanguageService {
+
+	List<String> languages();
+
+}
+
+@Component
+class LanguageServiceImpl implements LanguageService {
```

Comment:
Please make that a static class in the application itself. I don't like have several top-level classes in the same source file.